Source/Core/Logger/Logger.swift

```diff
@@ -0,0 +1,302 @@
+//
+// Copyright 2011 - 2018 Schibsted Products & Technology AS.
+// Licensed under the terms of the MIT license. See LICENSE in the project root.
+//
+
+import Foundation
+
+private extension DispatchTime {
+    var elapsed: Double {
+        let nanoTime = DispatchTime.now().uptimeNanoseconds - self.uptimeNanoseconds
+        return Double(nanoTime) / 1_000_000
+    }
+}
+
+/**
+ A logging class that can be told where to log to via transports.
+
+ Features include:
+ - Asynchronous, ordered output to transports
+ - All log messages can be tagged and filtered
+ - `LogLevel`s are provided as well
+ - Force logging enables output via `print` even if no transports available
+
+ ## Filtering
+
+ Two methods exist to allow for filtering of the log stream.
+
+ - `Logger.filterUnless(tag:)`
+ - `Logger.filterIf(tag:)`
+
+ */
+public class Logger {
+    /// Shared logger object
+    public static let shared: Logger = {
```

Comment:
You could just do `public static let shared = Logger()` if you don't need any extra setup.